### PR TITLE
John conroy/fix npm run test

### LINKED ISSUE #449: Move fetch polyfill for tests to setupTests

### LINKED ISSUE #862: Make mocks for js tests
(So `npm test` works.)

> We probably need a mock. So use `jest.js` test for now and I can triage this later.

_Originally posted by @john-conroy in https://github.com/hubmapconsortium/portal-ui/pull/859#issuecomment-662119947_

### PR DESCRIPTION
**Pull request by john-conroy**
Fixes #862. Fixes #449.

Disabling `Dataset.spec.js` was too messy so I opted to delete it.

### CODE REVIEW

File: context/jest.config.js [hunk 1]
```diff
@@ -2,11 +2,16 @@ module.exports = {
...
+    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
+      '<rootDir>/test-utils/__mocks__/file.js',
     '^.+\\.css$': '<rootDir>/test-utils/__mocks__/style.js',
     '^components(.*)$': '<rootDir>/app/static/js/components$1',
     '^static(.*)$': '<rootDir>/app/static/$1',
     '^test-utils(.*)$': '<rootDir>/test-utils$1',
     '^helpers(.*)$': '<rootDir>/app/static/js/helpers/$1',
+    '^hooks(.*)$': '<rootDir>/app/static/js/hooks/$1',
+    '^shared-styles(.*)$': '<rootDir>/app/static/js/shared-styles/$1',
+    '^theme$': '<rootDir>/app/static/js/theme.jsx',
```
Comment (mccalluc) [inline context/jest.config.js:14]: Not a blocker for now, but would it be possible to capture everything under `<rootDir>/app/static/js` without spelling out each subdirectory? I worry about someone reorganizing or adding a new directory down the road, and needing to remember to update this list.

Comment (john-conroy) [reply context/jest.config.js:14]: Yes, I definitely should. I'll clean things up before merging. I wanted to keep things parallel to our webpack config (which needs similar cleaning).

**Comment by mccalluc** [comment]
@john-conroy : Can this be merged?

**Comment by john-conroy** [comment]
I still need to clean up the mapped names/paths. Sorry, I've been focused elsewhere. I'll try and get it in tonight.

**Pull request merged**